src/snowflake/snowpark/modin/plugin/_internal/ordered_dataframe.py

```diff
@@ -1176,15 +1188,29 @@ def join(
             eq = Column(left_col).equal_null(Column(right_col))
             on = eq if on is None else on & eq
 
-        # If we are doing a cross join, `on` cannot be specified.
-        if how != "cross":
+        if how == "asof":
+            left_match_condition = Column(left_match_condition)
+            # Get the new mapped right match condition identifier
+            right_match_condition = Column(
+                right_identifiers_rename_map[right_match_condition]  # type: ignore
+            )
+            # ASOF Join requires the use of match_condition
             snowpark_dataframe = left_snowpark_dataframe_ref.snowpark_dataframe.join(
```

Comment:
Shouldn't `by` be specified here as the `on` parameter?

Reply:
I tried doing that, but there is some extra work required for `by/on` with ASOF Join. Specifically, I ended up getting this:

```
SnowparkSQLException: (1304): 01b67bf9-090b-0f68-000c-a905793aac47: 010010 (42601): SQL compilation error:
ON clause for ASOF JOIN must contain conjunctions of equality conditions only. Disjunctions are not allowed. Each side of an equality condition must only refer to either the left table or the right table. EQUAL_NULL(SNOWPARK_LEFT."ticker", SNOWPARK_RIGHT."ticker_ksr3") is invalid.
```

Reply:
Oh interesting - mind flagging that with the snowpark team? If that's the case, I agree that we don't need to support by here!